lib/type_check/internals/parser.ex

```diff
@@ -0,0 +1,322 @@
+defmodule TypeCheck.Internals.Parser do
+  @moduledoc """
```

Comment:
We do not want this module to show up in the generated public documentation, as this module is not intended for public usage but only for implementers,
so write `@moduledoc false` and move the current documentation to comments right below it.